sdk/lib/drivers/sound/mmebuddy/wave/streaming.c

```diff
@@ -206,6 +212,16 @@ CompleteIO(
             /* Partially completed */
             HdrExtension->BytesCompleted += dwNumberOfBytesTransferred;
             SND_TRACE(L"%d/%d bytes of wavehdr completed\n", HdrExtension->BytesCompleted, WaveHdr->dwBufferLength);
+
+            /* Do we loop a header? */
```

Comment:
Below, current function will execute
```c
    // completion callback is performed in a thread
    DoWaveStreaming(SoundDeviceInstance);
```
To confirm, does this comment mean there is no code loop, so no stack issue on "high" number of header loops?

Reply:
No. It's just remaining part of very old code, which was written before adding the loop (this `DoWaveStreaming` call). If you'll closely look to the commits history, you'll see the commit which added this:
https://git.reactos.org/?p=reactos-deprecated-gitsvn-dont-use.git;a=commitdiff;h=ca4df219f8e03f57f3721aba74e4d8da3748b940
(see streaming.c changes)
![изображение](https://github.com/reactos/reactos/assets/26385117/0dca53e2-6c82-41a9-872e-db862cb69697)
As I tested, this call is from old code, and it does not do anything useful. Since the header is already completed at this stage, after the loop is done, that call just fails on the header validation (e. g., it just prints "DoWaveStreaming: No work to do - doing nothing"). So it can be removed without any problems, but it is (and should be done) outside of (the context of) this PR I guess.

Reply:
Note that `FreeMemory(lpOverlapped);` must not be executed multiple times for the same memory.
I'll trust you that the above `DoWaveStreaming()` call always immediately returns.

> it can be removed without any problems, but it is (and should be done) outside of (the context of) this PR

I would feel safer if that was done (separately) before committing the current PR.